internal/pkg/service/stream/storage/level/local/diskwriter/network/transport/server.go

```diff
@@ -210,14 +211,19 @@ func (s *Server) acceptConnection(ctx context.Context, listener net.Listener) er
 		return err
 	}
 
+	if s.isClosed() {
+		_ = conn.Close()
+		return nil
+	}
+
 	// Create multiplexer
 	sess, err := yamux.Server(conn, multiplexerConfig(s.logger, s.config))
```

Comment:
I debuged the KCP test, with `-timeout 2s` flag, so I have seen, where it is blocked.

The key is to realize that the fact that the client has closed the connection does not mean that the server has already processed this information

So, after creating a connection/stream, we will check whether the server does not shut down, so that this connection/stream does not block the shutdown.

------------

We'll see if it's enough like this, it's a problem to simulate the issue locally.